system/density/buffers/buffer.c

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
I struggled with that - and finally decided to import the source to prevent newer versions with any possible backward compatibility / api changes causing issues.